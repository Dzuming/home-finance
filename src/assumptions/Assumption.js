import React, { Component } from 'react';
import Nav from './Nav';
import MonthlyAssumption from './MonthlyAssumption';
import OverallAssumptions from './OverallAssumptions';

const TABS = {
  0: () => <MonthlyAssumption />,
  1: () => <OverallAssumptions />,
};

class Assumption extends Component {
  state = {
    currentTab: 0,
  };

  handleTabChange = (event, value) => this.setState({ currentTab: value });

  render() {
    const { currentTab } = this.state;
    return (
      <React.Fragment>
        <Nav currentTab={currentTab} handleTabChange={this.handleTabChange} />
        {TABS[currentTab]()}
      </React.Fragment>
    );
  }
}

export default Assumption;
