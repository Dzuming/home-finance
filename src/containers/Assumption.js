import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Assumption extends Component {
  render() {
    return <div>test</div>;
  }
}

Assumption.propTypes = {};

export default Assumption;
