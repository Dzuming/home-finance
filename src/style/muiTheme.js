import { createMuiTheme } from 'material-ui/styles';

export const theme = createMuiTheme({
  status: {
    success: '#4CAF50',
    warning: '#F44336',
  },
});
