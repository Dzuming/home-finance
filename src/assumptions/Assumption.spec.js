import React from 'react';
import { create } from 'react-test-renderer';
import Assumption from './Assumption';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { theme } from '../style/muiTheme';
import { MuiThemeProvider } from 'material-ui';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({
  assumptions: {
    monthly: [],
  },
  user: [],
  login: { isAuthenticated: true },
});
const component = create(
  <MuiThemeProvider theme={theme}>
    <Provider store={store}>
      <Assumption />
    </Provider>,
  </MuiThemeProvider>,
);

describe('<Assumption />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
