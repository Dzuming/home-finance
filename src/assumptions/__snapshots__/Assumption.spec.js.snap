// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assumption /> should match snapshot 1`] = `
Array [
  <div
    className="MuiTabs-root-1"
  >
    <div
      className="MuiTabs-flexContainer-2"
    >
      <div
        className="MuiTabs-scrollingContainer-3 MuiTabs-fixed-4"
        onScroll={[Function]}
        role="tablist"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="MuiTabs-flexContainer-2"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root-24 MuiTab-root-8 MuiTab-rootInherit-10 MuiTab-rootInheritSelected-17"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            style={undefined}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper-20"
            >
              <span
                className="MuiTab-labelContainer-21"
              >
                <span
                  className="MuiTab-label-22"
                >
                  Monthly assumption
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-29"
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root-24 MuiTab-root-8 MuiTab-rootInherit-10"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            style={undefined}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper-20"
            >
              <span
                className="MuiTab-labelContainer-21"
              >
                <span
                  className="MuiTab-label-22"
                >
                  Overall assumptions
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-29"
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root-24 MuiTab-root-8 MuiTab-rootInherit-10"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            style={undefined}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper-20"
            >
              <span
                className="MuiTab-labelContainer-21"
              >
                <span
                  className="MuiTab-label-22"
                >
                  Add assumption
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-29"
            />
          </button>
        </div>
        <span
          className="MuiTabIndicator-root-26 MuiTabIndicator-colorSecondary-28"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-typeContainer-38 Connect-MonthlyAssumption--assumptionWrapper-37"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-128"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-138 MuiInputLabel-root-132 MuiInputLabel-formControl-133 MuiInputLabel-animated-136 MuiInputLabel-shrink-135"
        data-shrink={true}
        htmlFor="month"
      >
        assumption date
      </label>
      <div
        className="MuiInput-root-142 MuiInput-formControl-143 MuiInput-inkbar-144 MuiInput-underline-148"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-151 MuiInput-inputSingleline-154"
          defaultValue={undefined}
          disabled={false}
          id="month"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="month"
          value="2018-06"
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-typeContainer-38"
  >
    <div
      className="MuiGrid-typeItem-39 MuiGrid-grid-xs-3-66"
    >
      <div
        className="MuiPaper-root-158 CardList-card-157 MuiPaper-shadow2-162 MuiPaper-rounded-159"
      >
        <div
          className="MuiCardContent-root-185"
        >
          <aside
            className="MuiTypography-root-186 MuiTypography-body2-194 MuiTypography-colorInherit-205 MuiTypography-alignCenter-199"
          >
            Assumption
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-39 MuiGrid-grid-xs-3-66"
    >
      <div
        className="MuiPaper-root-158 CardList-card-157 MuiPaper-shadow2-162 MuiPaper-rounded-159"
      >
        <div
          className="MuiCardContent-root-185"
        >
          <aside
            className="MuiTypography-root-186 MuiTypography-body2-194 MuiTypography-colorInherit-205 MuiTypography-alignCenter-199"
          >
            Percentage
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-39 MuiGrid-grid-xs-3-66"
    >
      <div
        className="MuiPaper-root-158 CardList-card-157 MuiPaper-shadow2-162 MuiPaper-rounded-159"
      >
        <div
          className="MuiCardContent-root-185"
        >
          <aside
            className="MuiTypography-root-186 MuiTypography-body2-194 MuiTypography-colorInherit-205 MuiTypography-alignCenter-199"
          >
            Value
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-39 MuiGrid-grid-xs-3-66"
    >
      <div
        className="MuiPaper-root-158 CardList-card-157 MuiPaper-shadow2-162 MuiPaper-rounded-159"
      >
        <div
          className="MuiCardContent-root-185"
        >
          <aside
            className="MuiTypography-root-186 MuiTypography-body2-194 MuiTypography-colorInherit-205 MuiTypography-alignCenter-199"
          >
            Limit
          </aside>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
